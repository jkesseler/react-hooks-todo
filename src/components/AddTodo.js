import React, { memo, useContext, useState } from 'react';
import PropTypes from 'prop-types';

import {
  Button, Grid, Paper, TextField,
} from '@material-ui/core';

import Store from '../store';

export const AddTodo = ({
  todo,
  handleInputChange,
  handleKeyPress,
  handleTodoAdd,
}) => (
  <Paper>
    <Grid style={{ margin: 16, padding: 16 }} container>
      <Grid xs={10} md={11} style={{ paddingRight: 16 }} item>
        <TextField
          onChange={handleInputChange}
          onKeyPress={handleKeyPress}
          placeholder="Write something"
          value={todo}
          fullWidth
        />
      </Grid>
      <Grid xs={2} md={1} item>
        <Button
          color="secondary"
          variant="outlined"
          onClick={handleTodoAdd}
        >
          Add
        </Button>
      </Grid>
    </Grid>
  </Paper>
);


AddTodo.propTypes = {
  todo: PropTypes.string.isRequired,
  handleInputChange: PropTypes.func.isRequired,
  handleKeyPress: PropTypes.func.isRequired,
  handleTodoAdd: PropTypes.func.isRequired,
};


export default memo(() => {
  const { dispatch } = useContext(Store);
  const [todo, setTodo] = useState('');
  const handleInputChange = event => setTodo(event.target.value);

  const handleTodoAdd = () => {
    dispatch({ type: 'ADD_TODO', payload: todo });
    setTodo('');
  };

  const handleKeyPress = (event) => {
    if (event.which === 13 || event.keyCode === 13) {
      handleTodoAdd();
    }
  };

  const props = {
    todo,
    handleInputChange,
    handleKeyPress,
    handleTodoAdd,
  };

  return (<AddTodo {...props} />);
});
