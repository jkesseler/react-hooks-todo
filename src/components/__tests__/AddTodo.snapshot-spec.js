import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import unwrapMemo from '../../unwrap-memo';

import AddTodo from '../AddTodo';

describe('AddTodo', () => {
  it('renders without crashing', () => {
    const component = unwrapMemo(shallow, <AddTodo />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
