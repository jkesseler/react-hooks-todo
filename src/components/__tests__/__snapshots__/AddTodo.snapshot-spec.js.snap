// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo renders without crashing 1`] = `
<AddTodo
  handleInputChange={[Function]}
  handleKeyPress={[Function]}
  handleTodoAdd={[Function]}
  todo=""
/>
`;
